File: internal/graph/testdata/ts_simple/src/handler.ts
export class Handler {
  name: string;

  constructor(name: string) {
    this.name = name;
  }

  greet(): string {
    return this.format(this.name);
  }

  format(x: string): string {
    return "Hello, " + x;
  }
}

export function makeHandler(name: string): Handler {
  return new Handler(name);
}

File: internal/graph/testdata/rust_simple/src/handler.rs
pub trait Greeter {
    fn greet(&self) -> String;
}

pub struct Handler {
    pub name: String,
}

impl Handler {
    pub fn new(name: String) -> Self {
        Handler { name }
    }

    pub fn format(&self, x: &str) -> String {
        format!("Hello, {}", x)
    }
}

impl Greeter for Handler {
    fn greet(&self) -> String {
        self.format(&self.name)
    }
}

pub fn make_handler(name: String) -> Handler {
    Handler::new(name)
}
